plugins/tiddlywiki/multiwikiserver/modules/routes/handlers/manage-user.js

```diff
@@ -0,0 +1,80 @@
+/*\
+title: $:/plugins/tiddlywiki/multiwikiserver/routes/handlers/manage-user.js
+type: application/javascript
+module-type: mws-route
+
+GET /admin/users/:user_id
+
+\*/
+(function() {
+
+	/*jslint node: true, browser: true */
+	/*global $tw: false */
+	"use strict";
+	
+	exports.method = "GET";
+	
+	exports.path = /^\/admin\/users\/([^\/]+)\/?$/;
+	
+	exports.handler = function(request,response,state) {
+		var user_id = $tw.utils.decodeURIComponentSafe(state.params[0]);
+		var userData = state.server.sqlTiddlerDatabase.getUser(user_id);
+		
+		if(!userData) {
+			response.writeHead(404, "Not Found", {"Content-Type": "text/html"});
+			var errorHtml = $tw.mws.store.adminWiki.renderTiddler("text/plain", "$:/plugins/tiddlywiki/multiwikiserver/templates/error", {
+				variables: {
+					"error-message": "User not found"
+				}
+			});
+			response.write(errorHtml);
+			response.end();
+			return;
+		}
+		
+		// Check if the user is trying to access their own profile or is an admin
+		var hasPermission = ($tw.utils.parseInt(user_id, 10) === state.authenticatedUser.user_id) || state.authenticatedUser.isAdmin;
+		if(!hasPermission) {
+			response.writeHead(403, "Forbidden", { "Content-Type": "text/plain" });
+			response.end("Forbidden");
+			return;
+		}
+
+		// Convert dates to strings and ensure all necessary fields are present
+		var user = {
+			user_id: userData.user_id || "",
+			username: userData.username || "",
+			email: userData.email || "",
+			created_at: userData.created_at ? new Date(userData.created_at).toISOString() : "",
+			last_login: userData.last_login ? new Date(userData.last_login).toISOString() : ""
+		};
+	
+		// Get all roles which the user has been assigned
+		var userRole = state.server.sqlTiddlerDatabase.getUserRoles(user_id);
+		var allRoles = state.server.sqlTiddlerDatabase.listRoles();
+
+		// sort allRoles by placing the user's role at the top of the list
+		allRoles.sort(function(a, b){ (a.role_id === userRole.role_id ? -1 : 1) });
+		
+		response.writeHead(200, "OK", {
+			"Content-Type": "text/html"
+		});
+	
+		// Render the html
+		var html = $tw.mws.store.adminWiki.renderTiddler("text/plain", "$:/plugins/tiddlywiki/multiwikiserver/templates/page", {
+			variables: {
+				"page-content": "$:/plugins/tiddlywiki/multiwikiserver/templates/manage-user",
+				"user": JSON.stringify(user),
+				"user-initials": user.username.split(" ").map(name => name[0]).join(""),
+				"user-role": JSON.stringify(userRole),
+				"all-roles": JSON.stringify(allRoles),
+				"is-current-user-profile": state.authenticatedUser && state.authenticatedUser.user_id === $tw.utils.parseInt(user_id, 10) ? "yes" : "no",
```

Comment:
@webplusai - If you use `$tw.utils.parseInt(str)` there is no need to add the radix param. Since the utility function takes care of it. See: https://github.com/TiddlyWiki/TiddlyWiki5/blob/c7531e53abbe065d9d1a9fe9959f88b39d1a98f4/core/modules/utils/utils.js#L980-L982

Reply:
JFTR, you don't need it with the language's `parseInt` either, if you're calling it as a global function, where the radix defaults to `10`.  Of course if you're passing a reference to it into a function that calls it with additional parameters, you can be in for a surprise or two.  The classic example is:

```js
['1.1', '2.2', '3.3'].map(parseFloat) //=> [1.1, 2.2, 3.3]
// but
['1', '2', '3'].map(parseInt) //=> [1, NaN, NaN]
// hmm
```

That is not an issue, though with the current usage, either of the native version or the TW one.